CRM/CivirulesCronTrigger/RelationshipStartDate.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+class CRM_CivirulesCronTrigger_RelationshipStartDate extends CRM_Civirules_Trigger_Cron {
+
+  private $dao = false;
+
+  /**
+   * Returns an array of entities on which the trigger reacts
+   *
+   * @return CRM_Civirules_TriggerData_EntityDefinition
+   */
+  protected function reactOnEntity() {
+    return new CRM_Civirules_TriggerData_EntityDefinition('Relationship', 'relationship', 'CRM_Contact_DAO_Relationship', 'Relationship');
+  }
+
+  /**
+   * This method returns a CRM_Civirules_TriggerData_TriggerData this entity is used for triggering the rule
+   *
+   * Return false when no next entity is available
+   *
+   * @return object|bool CRM_Civirules_TriggerData_TriggerData|false
+   * @access protected
+   */
+  protected function getNextEntityTriggerData() {
+    if (!$this->dao) {
```

Comment:
Would there be a scenario where the DAO would be empty? Is this `queryForTriggerEntities` also in other cron trigger classes in the civirules extension?

Reply:
yes it is following the same approach as other triggers